.github/workflows/ci.yaml

```diff
@@ -0,0 +1,240 @@
+name: GitHub Actions CI
+
+# Cache is used for all Windows and macOS dependencies (size approx. 1230 * 2 + 1850 = 4310 MiB)
+# Cache is not used for Ubuntu builds, because it already has all dependencies except
+# the appropriate libtorrent version, which only takes 3-5 minutes to build from source anyway
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    types: [edited, opened, reopened, synchronize]
+    branches: [ master ]
+
+env:
+  # Qt: 5.15.0
+  # libtorrent: RC_1_2 HEAD, 1.2.10
+  VCPKG_COMMIT: 32eccc18191fbb57b159784a1724d2d00613ae82
+  VCPKG_DEST_MACOS: /Users/runner/qbt_tools/vcpkg
+  VCPKG_DEST_WIN: C:\qbt_tools\vcpkg
+  UBUNTU_LIBTORRENT_VERSION: libtorrent-1.2.10
+
+# NOTE: for client code built with >= C++14, libtorrent 1.2.x must be built with >= C++14 as well.
+# Compilers that don't default to >= C++14 must be overriden.
+# See https://github.com/arvidn/libtorrent/issues/4965.
+
+jobs:
+
+  ci_ubuntu:
+    name: Ubuntu
+
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, ubuntu-18.04]
+        qbt_gui: ["GUI=ON", "GUI=OFF"]
+      fail-fast: false
+
+    runs-on: ${{ matrix.os }}
+
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+    - name: checkout repository
+      uses: actions/checkout@v2.3.2
+
+    - name: install all build dependencies except libtorrent from Ubuntu repos
+      run: |
+        sudo apt install \
+          build-essential cmake git ninja-build pkg-config \
+          libssl-dev libgeoip-dev zlib1g-dev \
+          libboost-dev libboost-chrono-dev libboost-random-dev libboost-system-dev \
+          libqt5svg5-dev qtbase5-dev qttools5-dev
+
+    - name: install libtorrent from source
+      run: |
+        git clone https://github.com/arvidn/libtorrent && cd libtorrent
+        git checkout ${{ env.UBUNTU_LIBTORRENT_VERSION }}
+        cmake -B cmake-build-dir -G "Ninja" \
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+          -Ddeprecated-functions=OFF \
+          --graphviz=cmake-build-dir/target_graph.dot
+        cmake --build cmake-build-dir
+        sudo cmake --install cmake-build-dir --prefix /usr/local
+
+    - name: build qBittorrent
+      run: |
+        cmake -B build -G "Ninja" \
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+          -D${{ matrix.qbt_gui }} \
+          -DVERBOSE_CONFIGURE=ON \
+          --graphviz=build/target_graph.dot
+        cmake --build build
+
+    - name: upload artifact as zip
+      uses: actions/upload-artifact@v2.1.3
```

Comment:
@Chocobo1 

That's not good practice, it has the potential to break our CI and will be difficult to track down - one would have to check recent updates to _all_ actions to figure out what changed. It is preferable to bump these every now and then. The CI itself will catch any breakage due to specific version bumps, and the whole situation becomes much easier to investigate and resolve, since we'll know immediately what changed.

Reply:
>It is preferable to bump these every now and then. 

This is something I wouldn't prefer, especially our codebase is not for CI itself.

@glassez 
Any opinion on this?

Reply:
@Chocobo1 

> This is something I wouldn't prefer, especially our codebase is not for CI itself.

This extreme aversion to commits that relate to ancillary tasks not directly related to source code changes (let me guess, such bumps "pollute the git history") is starting to seriously bother me. 
No, it does not make "the codebase only about the CI itself". These kinds of changes are as much part of the guaranteeing smooth sailing of the project as much as anything else. They **are** part of the project development.

I don't want to have "fun" trying to figure out which Action version bump completely broke or changed something about our CI each time in the future, and neither do you for sure.

If you have concerns that the git history has or will become too difficult to sort through, the proper course of action is to find a proper solution for that, such as agreed-upon commit prefixes similar to what https://github.com/doitsujin/dxvk and other projects do, not fight every single proposed change with the "git history pollution" argument. Hell, a lot of projects have bots automatically committing translation updates and other things and they do just fine.

Reply:
>This extreme aversion to commits concerning ancillary tasks not directly related to source code changes (let me guess, such bumps "pollute the git history") is starting to seriously bother me.

Who knows how many minor version bumps for CI script are going to happen once this PR is merged? Of course this is a major concern for the developers.

>I don't want to have "fun" trying to figure out which Action version bump completely broke or changed something about our CI each time in the future, and neither do you for sure.

I'm OK with explicit versions for vcpkg scripts under only one condition: don't **ever** change it if it ain't broke or without good reasoning.
If you think that can uphold, then I have no issue with them.

Reply:
@Chocobo1 

>Who knows how many minor version bumps for CI script are going to follow once this PR is merged? Of course this is a major concern for the developers.

As little as possible, and as "in bulk" as possible.

>I'm OK with explicit versions for vcpkg scripts under only one condition: don't ever change it if it ain't broke or without good reasoning.
If you think that can uphold, then I have no issue with them

We are in agreement then.